Update DataFrame schema returned from `mlflow.search_trace()` to be V3 format (#15643)
mlflow/mlflow: docs/docs/tracing/api/search.mdx

    - **state**: State of the trace, can be one of [`OK`, `ERROR`, `IN_PROGRESS`, `STATE_UNSPECIFIED`].
    - **request_time**: The start time of the trace in milliseconds
    - **execution_duration**: The duration of the trace in milliseconds
    - **request**: The input to the traced logic
    - **response**: The output of the traced logic
    - **trace_metadata**: Key-value pairs associated with the trace
    - **spans**: List of spans in the trace
    - **tags**: List of assessments associated with the trace.
  </TabItem>
  <TabItem value="2.x" label="MLflow 2.x">
    - **request_id**: A primary identifier of a trace
    - **trace**: A trace object
    - **timestamp_ms**: The start time of the trace in milliseconds
    - **status**: The status of the trace
    - **execution_time_ms**: The duration of the trace in milliseconds
    - **request**: The input to the traced logic
    - **response**: The output of the traced logic
    - **request_metadata**: Key-value pairs associated with the trace
    - **spans**: Spans in the trace
    - **tags**: Tags associated with the trace
  </TabItem>
</Tabs>

:::note

> The format of the DataFrame depends on the MLflow version when you call the `search_traces` API,

Reviewer: ```suggestion
The format of the DataFrame depends on the MLflow version used to call the `search_traces` API,
```
